src/Common/tests/System/Net/VirtualNetwork/VirtualNetworkStream.cs

```diff
@@ -92,16 +105,15 @@ public override void Write(byte[] buffer, int offset, int count)
             _network.WriteFrame(_isServer, innerBuffer);
         }
 
-        private void UpdateReadStream()
+        public override Task<int> ReadAsync(byte[] buffer, int offset, int count, CancellationToken cancellationToken)
         {
-            if (_readStream != null && (_readStream.Position < _readStream.Length))
-            {
-                return;
-            }
+            return Task.FromResult<int>(this.Read(buffer, offset, count));
+        }
 
-            byte[] innerBuffer;
-            _network.ReadFrame(_isServer, out innerBuffer);
-            _readStream = new MemoryStream(innerBuffer);
+        public override Task WriteAsync(byte[] buffer, int offset, int count, CancellationToken cancellationToken)
+        {
+            this.Write(buffer, offset, count);
```

Comment:
No, the idea is to overload WriteAsync so it doesn't go through the base (Stream) implementation. Since no actual I/O is performed, the operation is completed as for Write, synchronously and a completed Task returned.